### ISSUE TITLE
Calendar isn't rendering if browser reload (video)

### ISSUE THREAD

**Issue by vapits** [angular-ui/ui-calendar#262]
Hi,
it seems that the calendar isn't renders if i reload/refresh the browser.
Other wize it works like a charm.

Here's a video example:
https://www.youtube.com/watch?v=GMVdNAwQq7g

Heres the html:
````html
<div ui-calendar="uiConfig.calendar" ng-model="eventSources"></div>
````

and here's the controller:

````javascript
dashBoard.controller('powerToolsCtrl', ['$scope', 'uiCalendarConfig', function($scope, uiCalendarConfig){
   $scope.events = [
    {
      title: 'Event @ Roof Garden',
      start: '2015-05-02T20:00:00',
      end: '2015-05-02T24:00:00',
      allDay: false
    },
    {
      title: 'Two Days Event @ Pool',
      start: '2015-05-03T08:00:00',
      end: '2015-05-04T10:00:00'
    },
    {
      title: 'Offer',
      start: '2015-05-23T08:00:00+08:00',
      end: '2015-05-31T10:00:00+08:00'
    }
  ];


  $scope.uiConfig = {
      calendar:{
        height: 450,
        editable: false,
        header:{
          left: 'title',
          center: '',
          right: 'today prev,next'
        },
        dayClick: $scope.alertEventOnClick,
        eventDrop: $scope.alertOnDrop,
        eventResize: $scope.alertOnResize,
        defaultView : 'month'
      }
    };

    $scope.eventSources = [{ events: $scope.events, color:'purple', textColor:'white' }];
}]);
````

**Comment by LilBiggs** [comment]
@vapits,

Any errors in the console?

**Comment by vapits** [comment]
No not all.

**Comment by LilBiggs** [comment]
Created a plunker and it works fine. Must be some other code you have causing the issue. If you can recreate it on plunker that would help us debug.

[Plunker] (http://plnkr.co/edit/QwVPBL?p=info)

**Comment by joshkurz** [comment]
thanks @LilBiggs, yes please recreate this issue in a plunker @vapits.

**Comment by lodnert** [comment]
@vapits dit you find any solution to this, i have the same error. I'm testing out UI-calender with Ui-router i think that where my problem is.

Any ideas?

**Comment by jimmyparekh** [comment]
ui calendar not working with the mozilla firefox. can u please help for that

**Comment by alexander-clark** [comment]
Struggled with this same issue, but eventually found a fix.

By putting a bunch of `console.log()`s in the source of source of fullCalendar itself, I discovered that the difference between navigating with a ui-sref and doing a browser reload was that during a reload, `elementVisible()` was returning false in `renderView()`, preventing the view from getting rendered.

Based on an [article I found](http://blog.brunoscopelliti.com/run-a-directive-after-the-dom-has-finished-rendering/), I wrapped the `$scope.uiConfig = {}` section in a `$timeout()`. 100ms seemed to be the magic number to get it to work. Not sure why, but it's sub-second so I won't complain too much.

```javascript
$timeout(function() {
  $scope.uiConfig = {
    calendar: {
      defaultView: 'agendaDay'
    }
  };
}, 100);
```

This seemed kind of ugly and vaguely familiar, so I tracked down #397 and tried the solution there. I had tried that before, and indeed, it still didn't quite work until I added the 100ms delay:

```javascript
setTimeout(function () {
    calendar.fullCalendar(options);
}, 100);
```

This allowed me to remove the $timeout hack.

For what it's worth, I'm using ui-router and angular-material, two things people have mentioned as potential culprits...
